specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1097 @@
+    condition which are not reported.)
+
+1.  Non-`Ready` conditions with non-error severity MAY be surfaced by the
+    implementation. Examples of `Warning` or `Info` conditions could include:
+    missing health check definitions, scale-to-zero status, or non-fatal
+    capacity limits.
+
+Conditions type names should be chosen to describe positive conditions where
+`True` means that the condition has been satisfied. Some conditions may be
+transient (for example, `ResourcesAllocated` might change between `True` and
+`False` as an application scales to and from zero). It is RECOMMENDED that
+transient conditions be indicated with a `severity="Info"`.
+
+# Resource Lifecycle
+
+## Broker Lifecycle
+
+A Broker represents an Addressable endpoint (i.e. it has a `status.address.url`
+field) which can receive, store, and forward events to multiple recipients based
+on a set of attribute filters (Triggers). Triggers are associated with a Broker
+based on the `spec.broker` field on the Trigger; it is expected that the
+controller for a Broker will also control the associated Triggers. When the
+Broker's `Ready` condition is `true`, the Broker MUST provide a
+`status.address.url` which accepts all CloudEvents and MUST forward the received
+events to each associated Trigger whose `Ready` condition is `true`. As
+described in the [Trigger Lifecycle](#trigger-lifecycle) section, a Broker MAY
+forward events to an associated Trigger which which does not currently have a
+`true` `Ready` condition, including events received by the Broker before the
+Trigger was created.
+
+When a Broker is created, its `spec.class` field MUST be populated to indicate
+which of several possible Broker implementations to use. It is RECOMMENDED to
+default the `spec.class` field on creation if it is unpopulated. Once created,
+the `spec.class` field MUST be immutable; the Broker must be deleted and
+re-created to change the `spec.class`. This pattern is chosen to make it clear
+that changing `spec.class` is not an atomic operation and that any
+implementation would be likely to result in message loss during the transition.
+
+## Trigger Lifecycle
+
+The lifecycle of a Trigger is independent of the Broker it refers to in its
+`spec.broker` field; if the Broker does not currently exist or the Broker's
+`Ready` condition is not `true`, then the Trigger's `Ready` condition MUST be
+`false`, and the reason SHOULD indicate that the corresponding Broker is missing
+or not ready.
+
+The Trigger MUST also set the `status.subscriberUri` field based on resolving
+the `spec.subcriber` field before setting the `Ready` condition to `true`. If
+the `spec.subscriber.ref` field points to a resource which does not exist or
+cannot be resolved via [Addressable resolution](#addressable-resolution), the
+Trigger MUST set the `Ready` condition to `false`, and at least one condition
+should indicate the reason for the error.
```

Comment:
Just jotting this down so I don't forget, but we should also mention about the Dependency Annotation errors.
Specifying the annotation: knative.dev/dependency

This never made it to spec :(